scikit-hep/boost-histogram
# Class families

Adding class families.

* New families of classes can be defined with `@set_family`.
    * The `cpp` classes are now distinct, with reprs directly from C++'s `<<`.
    * The `cpp` axes are now distinct classes with C++ calling conventions, closes #207
    * Test added for external use of `@set_family`
* The lookup system for C++ classes is now unified, with a single `@register(class_list)` and `cast` instead of a collection of `_to_*` functions and various `._CLASSES` and such. If there is just one C++ class associated (through subclassing) to a Python class, `@register` is not needed.
* `__module__` is now being set to the correct locations using `@set_module`
* `__all__` has been added to public modules. It is always the third line, after the future stuff.
* Free functions mentioned in #197 dropped (closes #197)
* Python mode reprs respect Python class names and calling conventions
* Add Transforms: Log, Sqrt, and Pow
    * Remove Regular.pow, etc. shortcuts.
* Nicer error when C++ module is not built (like when running in main directory)

Note that accumulators currently do not participate in the family scheme.
